app/state.go

```diff
@@ -448,7 +450,7 @@ type updateCommitState struct {
 	reportTries int
 }
 
-func NewUpdateCommitState(update *datastore.UpdateInfo) State {
+func NewUpdateCommitState(update *datastore.UpdateInfo) UpdateState {
```

Comment:
Return `UpdateState` as then the `controlMap` wrapper states, don't need to store the update themselves.
This should be strictly transparent, as `UpdateState` is a super state of `State`, and hence, can be passed in as `State` anywhere it is required, so this change will not change any existing logic.